src/main.ts

```diff
@@ -129,10 +164,14 @@ async function run() {
 
         core.info(`Downloading ${project} from ${url}`)
         const binPath = await tc.downloadTool(url);
-        const extractedPath = await extractFn(binPath);
-        core.info(`Successfully extracted ${project} to ${extractedPath}`)
+        await extractFn(binPath, dest);
 
-        core.addPath(extractedPath);
+        if (cacheEnabled && cacheKey !== undefined) {
+            await cache.saveCache([dest], cacheKey);
+        }
+
+        core.addPath(dest);
```

Comment:
There is a change here even when not opting into the cache, the path where the tool is installed is changing from a random UUID path to a path under hostedtoolcache (RUNNER_TOOL_CACHE) based on owner, repo, version, architecture  (`dest` var). I think this should be ok. `hostedtoolcache` is not preserved across GHA worker runs but may be preserved across self-hosted runs.